include/crocoddyl/multibody/costs/contact-cop-position.hpp

```diff
@@ -0,0 +1,148 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2020, University of Duisburg-Essen, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_MULTIBODY_COSTS_CONTACT_COP_POSITION_HPP_
+#define CROCODDYL_MULTIBODY_COSTS_CONTACT_COP_POSITION_HPP_
+
+#include "crocoddyl/multibody/fwd.hpp"
+#include "crocoddyl/multibody/cost-base.hpp"
+#include "crocoddyl/multibody/contact-base.hpp"
+#include "crocoddyl/multibody/contacts/contact-3d.hpp"
+#include "crocoddyl/multibody/contacts/contact-6d.hpp"
+#include "crocoddyl/multibody/data/contacts.hpp"
+#include "crocoddyl/multibody/frames.hpp"
+#include "crocoddyl/multibody/data/multibody.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+
+namespace crocoddyl {
+
+template <typename _Scalar>
+class CostModelContactCoPPositionTpl : public CostModelAbstractTpl<_Scalar> {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef CostModelAbstractTpl<Scalar> Base;
+  typedef CostDataContactCoPPositionTpl<Scalar> Data;
+  typedef StateMultibodyTpl<Scalar> StateMultibody;
+  typedef CostDataAbstractTpl<Scalar> CostDataAbstract;
+  typedef ActivationModelAbstractTpl<Scalar> ActivationModelAbstract;
+  typedef ActivationModelQuadTpl<Scalar> ActivationModelQuad;
+  typedef DataCollectorAbstractTpl<Scalar> DataCollectorAbstract;
+  typedef FrameCoPSupportTpl<Scalar> CoPSupport;
+  typedef typename MathBase::Vector2s Vector2s;
+  typedef typename MathBase::Vector3s Vector3s;
+  typedef typename MathBase::VectorXs VectorXs;
+  typedef typename MathBase::MatrixXs MatrixXs;
+  typedef typename MathBase::MatrixX3s MatrixX3s;
+  typedef Eigen::Matrix<Scalar, 4, 6> Matrix46;
+
+  CostModelContactCoPPositionTpl(boost::shared_ptr<StateMultibody> state,
+                          boost::shared_ptr<ActivationModelAbstract> activation, const CoPSupport& cop_support, 
+                          const Vector3s& normal, const std::size_t& nu);
+  virtual ~CostModelContactCoPPositionTpl();
+
+  virtual void calc(const boost::shared_ptr<CostDataAbstract>& data, const Eigen::Ref<const VectorXs>& x,
+                    const Eigen::Ref<const VectorXs>& u);
+  virtual void calcDiff(const boost::shared_ptr<CostDataAbstract>& data, const Eigen::Ref<const VectorXs>& x,
+                        const Eigen::Ref<const VectorXs>& u);
+  virtual boost::shared_ptr<CostDataAbstract> createData(DataCollectorAbstract* const data);
+
+  const CoPSupport& get_copSupport() const;
```

Comment:
Remove this function. Instead we need to define `set_referenceImpl`. See this example: https://github.com/loco-3d/crocoddyl/blob/master/include/crocoddyl/multibody/costs/contact-friction-cone.hxx#L100

Reply:
We need the frame id for getting the `contact`, currently we use `get_copSupport()` to have access to this id. How do you want to retrieve the frame id then?

Reply:
Ok, I forgot to mention that we need to define `get_referenceImpl` as well.
Is this answer your question?

Reply:
Actually not. I don't understand how `get_referenceImpl` allows for accessing the `FrameCoPSupport` values since there seem to be no return values. What do I miss here?